lib/plugins/removeNullCharsPlugin.ts

```diff
@@ -0,0 +1,36 @@
+import type { FastifyPluginCallback } from 'fastify'
+import fp from 'fastify-plugin'
+
+const removeNullChars = (data: unknown): unknown => {
+  if (typeof data === 'string') {
+    return data.replace(/\u0000/g, '');
+  }
+
+  if (Array.isArray(data)) {
+    return data.map(removeNullChars);
+  }
+
+  if (data !== null && typeof data === 'object') {
+    return Object.fromEntries(
+        Object.entries(data).map(([key, value]) => [key, removeNullChars(value)])
+    );
+  }
+
+  return data;
+}
+
+const pluginCallback: FastifyPluginCallback = (fastify, _options, done) => {
+  fastify.addHook('preValidation', (request, reply, done) => {
```

Comment:
I would suggest exposing hook as a standalone thing as well, so that it could be registered as preHandler on specific routes